extension/src/experiments/model/filterBy/index.ts

```diff
@@ -105,12 +107,31 @@ const evaluate = <T extends string | number | boolean>(
   }
 }
 
+const evaluateTagsFilter = (
+  experiment: Experiment,
+  filter: FilterDefinition
+) => {
+  const values = experiment.commit?.tags
+  if (!values || values.length === 0) {
+    return true
+  }
+  if (Array.isArray(values)) {
+    return !values.some(value =>
+      evaluate<typeof value>(value, filter.operator, filter.value as string)
+    )
+  }
+}
+
 const buildFilter =
   (
     filterDefinitions: FilterDefinition[]
   ): ((experiment: Experiment) => boolean) =>
   experiment => {
```

Comment:
I noticed that experiments will always be hidden since they can't have tags. Is the behavior expected or should we be showing the experiments if their parent commit has a right tag?

Reply:
The user seems to be happy with what has been implemented: https://github.com/iterative/vscode-dvc/issues/4878#issuecomment-1778510087. Let's release this and see what feedback we get.